rewrite-java/src/main/java/org/openrewrite/java/JavaTemplate.java

```diff
@@ -247,13 +246,85 @@ public J visitMethodDeclaration(J.MethodDeclaration method, Integer p) {
+                                        for(J.TypeParameter typeParameter : method.getTypeParameters()) {
+                                            J.Identifier typeParamIdent = (J.Identifier) typeParameter.getName();
+                                            if(typeParamIdent.getSimpleName().equals(typeParameterName)) {
+                                                List<TypeTree> bounds = typeParameter.getBounds();
+                                                JavaType.FullyQualified bound;
+                                                if(bounds.size() == 0) {
+                                                    bound = JavaType.Class.OBJECT;
+                                                } else {
+                                                    bound = (JavaType.FullyQualified) bounds.get(0);
+                                                }
+                                                JavaType.GenericTypeVariable genericType = new JavaType.GenericTypeVariable(
+                                                        typeParamIdent.getSimpleName(),
+                                                        bound);
+
+                                                paramTypes.add(genericType);
+                                            }
+                                        }
+                                    } else {
+                                        paramTypes.add(namedVariable.getType());
+                                    }
+                                }
+
+                                newType = newType.withParamNames(paramNames)
+                                        .withGenericSignature(newType.getGenericSignature().withParamTypes(paramTypes))
+                                        .withResolvedSignature(newType.getResolvedSignature().withParamTypes(paramTypes));
+                            }
+
+                            return method.withParameters(parameters)
+                                    .withType(newType);
                         }
                         case THROWS: {
                             J.MethodDeclaration m = method.withThrows(substitutions.unsubstitute(templateParser.parseThrows(substitutedTemplate)));
 
+                            // Update method type information to reflect the new checked exceptions
+                            JavaType.Method type = m.getType();
+                            if(type != null) {
+                                List<JavaType.FullyQualified> newThrows = new ArrayList<>();
+                                for(NameTree t : m.getThrows()) {
+                                    J.Identifier exceptionIdent = (J.Identifier) t;
+                                    newThrows.add((JavaType.FullyQualified) exceptionIdent.getType());
+                                }
+                                type.withThrownExceptions(newThrows);
```

Comment:
thinking this wa meant to be `type = type.withThrownExceptions(newThrows);`

Same thing here, probably should be going through JavaType.Method.build() to take advantage of flyweights

Reply:
Good catch! I didn't notice because I also didn't finish the corresponding test case - but in such a way that it passed with this code incomplete. 😅